### ISSUE TITLE
Cannot ignore rule jinja[spacing]

### ISSUE THREAD

**Issue by AdrienEti** [ansible/ansible-lint#3935]
##### Summary
The  [Jinja white-space control](https://jinja.palletsprojects.com/en/3.1.x/templates/#whitespace-control)  generates an jinja [spacing] rule violation. 
I'm trying to escape this rule for a variable definition, but it seems that the escape is ignored. 
I've tried to place the # noqa: jinja[spacing] mutation in several places, but it is always ignored.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
$ ansible-lint --version
ansible-lint 6.22.1 using ansible-core:2.16.0 ansible-compat:4.1.10 ruamel-yaml:0.18.5 ruamel-yaml-clib:0.2.7
```

- ansible installation method: **pip**
- ansible-lint installation method: **pip**

##### STEPS TO REPRODUCE

Defined a variable as follow and try to ignore the jinja[spacing] rule:
```yaml
foo_postgresql_grafana_password_encoded: >- # noqa: jinja[spacing]
  {% if '!' in foo_postgresql_grafana_password or '#' in foo_postgresql_grafana_password %}
  {{- ('\"\"\"' + foo_postgresql_grafana_password + '\"\"\"') | b64encode -}}
  {% else %}
  {{- foo_postgresql_grafana_password | b64encode -}}
  {% endif %}
``` 
or

```yaml
# noqa: jinja[spacing]
foo_postgresql_grafana_password_encoded: >-
  {% if '!' in foo_postgresql_grafana_password or '#' in foo_postgresql_grafana_password %}
  {{- ('\"\"\"' + foo_postgresql_grafana_password + '\"\"\"') | b64encode -}}
  {% else %}
  {{- foo_postgresql_grafana_password | b64encode -}}
  {% endif %}
``` 

##### Desired Behavior

The jinja[spacing] rule is ignored 

##### Actual Behavior

The jinja[spacing] rule is not ignored

**Comment by s3lph** [comment]
I'm affected by the same issue in exactly the same situation. I've found a workaround: Use `|-` instead of `>-`, and make all jinja tags remove whitespace (i.e. use `{%- -%}`  rather than `{% %}`. That way, whitespace is still removed, but by Jinja rather than by the YAML parser.

So, in your example:
```yaml
foo_postgresql_grafana_password_encoded: |-
  {%- if '!' in foo_postgresql_grafana_password or '#' in foo_postgresql_grafana_password -%}
  {{- ('\"\"\"' + foo_postgresql_grafana_password + '\"\"\"') | b64encode -}}
  {%- else -%}
  {{- foo_postgresql_grafana_password | b64encode -}}
  {%- endif -%}
```

**Comment by n0one42** [comment]
I've encountered a similar problem regarding YAML linting rules. While the following directive successfully disables the 'colons' rule:
```
# yamllint disable rule:colons
```

 I've been unable to disable the 'jinja[spacing]' rule using either of these approaches: 
```
# yamllint disable rule:jinja[spacing]
# noqa: jinja[spacing]
```
 Neither of these directives appears to have the desired effect on the 'jinja[spacing]' rule.
 
ansible core version: 2.17.4
ansible-lint version: 24.9.2
python version: 3.12.7

**Comment by StopMotionCuber** [comment]
I'm also encountering this issue with no apparent fix for it
